RockPaperScissors/RockPaperScissors/main.swift

```diff
@@ -1,10 +1,115 @@
 //
 //  RockPaperScissors - main.swift
-//  Created by yagom. 
+//  Created by Doogie, Eddy
 //  Copyright © yagom academy. All rights reserved.
-// 
+//
 
 import Foundation
 
-print("Hello, World!")
+enum GameOptions {
+    case exit
+    case scissor
+    case rock
+    case paper
+    case error
+    init(option: String) {
+        switch option {
+        case "0":
+            self = .exit
+        case "1":
+            self = .scissor
+        case "2":
+            self = .rock
+        case "3":
+            self = .paper
+        default:
+            self = .error
+        }
+    }
+}
+
+enum GameResult {
+    case exit
+    case win
+    case draw
+    case lose
+    case error
+}
+
+enum ResultTexts {
+    static let winText = "이겼습니다!"
+    static let drawText = "비겼습니다!"
+    static let loseText = "졌습니다!"
+    static let endText = "게임 종료"
+    static let errorText = "잘못된 입력입니다. 다시 시도해주세요."
+    static func printWin() {
+        print(winText)
+        print(endText)
+    }
+    static func printDraw() {
+        print(drawText)
+        showMenu()
+    }
+    static func printlose() {
+        print(loseText)
+        print(endText)
+    }
+    static func printEnd() {
+        print(endText)
+    }
+    static func printError() {
+        print(errorText)
+        showMenu()
+    }
+}
+
+func showMenu() {
+    print("""
+가위(1), 바위(2), 보(3)! <종료 : 0> :
+""", terminator: " ")
+    showResult(compare(userOption: matchedHand(receiveNumber()), computerOption: matchedHand(makeRandomNumber())))
```

Comment:
이 부분은 길어서 가독성이 떨어지는 것 같습니다.
두가지 방법으로 개선해볼 수 있을 것 같아요.
* 줄바꿈을 해서 가독성 개선
* 파라미터에 default 값을 할당해줘서 개선 ([Default Parameter Values](https://docs.swift.org/swift-book/LanguageGuide/Functions.html#ID166))
어떻게 개선해볼지는 여러분들의 선택입니다.

줄바꿈의 예시에요. 
```suggestion
    showResult(
        compareStepOne(
            userOption: matchedHand(receiveNumber()),
            computerOption: matchedHand(makeRandomNumber())
        )
    )
```